doc/development/style_guide.md

```diff
@@ -92,6 +92,38 @@ If using pre-commit is not an option, for whatever reason, there is a helper
 script [grass_clang_format.sh](./utils/grass_clang_format.sh), which simplifies
 bulk reformatting.
 
+#### Order of include headers
+
+In general, headers should be included in the order:
+
+1. Core system headers (stdio.h, ctype.h, ...)
+2. Headers for non-core system components (X11, libraries).
+3. GRASS headers (grass/gis.h, grass/glocale.h, ...)
+4. Headers for the specific library/program (geodesic.h, ...)
+
+Each class of headers has an obligation to be compatible with those above it in
+the list, but not those below it. The header groups should be alphabetically
+sorted and separated by a newline.
```

Comment:
> The header groups should be alphabetically sorted and separated by a newline.

Good you included (pun intended :) this!

(This is useful if we may choose to format all files one day, in batches preferably, to be able to add SortIncludes setting for ClangFormat).